docs/source/getting-started/app-architecture.rst

```diff
@@ -50,12 +57,16 @@ The last thing we'll point out about the DAML model for now is the operation to
   :end-before: -- ADDFRIEND_END
 
 DAML contracts are *immutable* (can not be changed in place), so the only way to "update" one is to archive it and create a new instance.
-That is what the ``AddFriend`` choice does: after checking some preconditions, it creates a new user contract with the new friend added to the list.
-The ``choice`` syntax automatically includes the archival of the current instance.
+That is what the ``AddFriend`` choice does: after checking some preconditions, it archives the current user contract and creates a new instance with the new friend added to the list.
```

Comment:
This also mixes up `contract` and `instance`. I think it would be easier to understand if you just use `contract` everywhere and at the beginning mention that a contract is an instance of a template.

Reply:
Agreed.